parsers/OPENNEM.py

```diff
@@ -0,0 +1,142 @@
+import arrow
+import datetime
+import json
+import logging
+import requests
+
+import pandas as pd
+import numpy as np
+
+ZONE_KEY_TO_REGION = {
+    'AUS-NSW': 'nem/nsw1',
+    'AUS-QLD': 'nem/qld1',
+    'AUS-SA': 'nem/sa1',
+    'AUS-TAS': 'nem/tas1',
+    'AUS-VIC': 'nem/vic1',
+    'AUS-WA': 'wem',
+}
+SOURCE = 'opennem.org'
+
+
+def dataset_to_df(dataset):
+    series = dataset['history']
+    interval = series['interval']
+    dt_i = arrow.get(series['start']).datetime
+    dt_f = arrow.get(series['last']).datetime
+    _type = dataset['data_type']
+    _id = dataset.get('id', 'price')  # price doesn't have an id
+
+    if _type != 'power':
+        name = _type.upper()
+    else:
+        name = _id.split(".")[-2].upper()
+
+    # Turn into minutes
+    if interval[-1] == "m":
+        interval += "in"
+
+    index = pd.date_range(start=dt_i, end=dt_f, freq=interval)
+    assert len(index) == len(series['data'])
+    df = pd.DataFrame(index=index, data=series['data'], columns=[name])
+    return df
+
+
+def fetch_main_df(zone_key=None, session=None, target_datetime=None, logger=logging.getLogger(__name__)):
+    region = ZONE_KEY_TO_REGION[zone_key]
+
+    if target_datetime:
+        # We will fetch one week in the past
+        df_start = arrow.get(target_datetime).shift(days=-7).datetime
+        y, w, d = df_start.isocalendar()
+        iso_week = "{0}W{1:02d}".format(y, w)
+        raise Exception('Not tested yet')
```

Comment:
Hey Oli I don't know if this is intentionally left not working?

Reply:
Ping @corradio